src/mantine-core/src/components/Select/Select.tsx

```diff
@@ -202,6 +206,7 @@ export const Select = forwardRef<HTMLInputElement, SelectProps>(
       cancelable: false,
       isList: true,
     });
+    const isDeselectedValue = useRef<boolean>(false);
```

Comment:
There is no need to keep deselected state, dropdown should close when item is clicked – if user deselects value it means that the user does not want it and thus dropdown should be closed